File: FamlyAndroidEdition/app/src/main/java/com/mack/john/famly_androidedition/objects/account/Account.java
package com.mack.john.famly_androidedition.objects.account;

import android.graphics.Bitmap;
import android.graphics.drawable.Drawable;

import com.mack.john.famly_androidedition.objects.account.profile.Profile;

import java.io.Serializable;
import java.util.ArrayList;

public class Account implements Serializable {



    // Class properties
    String familyName;
    String streetAddress;
    int postalCode;
    String masterEmail;
    String masterPassword;
    ArrayList<Profile> profiles;



    // Constructor
    public Account(String familyName, String streetAddress, int postalCode, String masterEmail, String masterPassword) {
        this.familyName = familyName;
        this.streetAddress = streetAddress;
        this.postalCode = postalCode;
        this.masterEmail = masterEmail;
        this.masterPassword = masterPassword;
        this.profiles = new ArrayList<>();
    }




    // Getters / Setter
    public String getFamilyName() {
        return familyName;
    }

    public void setFamilyName(String familyName) {
        this.familyName = familyName;
    }

    public String getStreetAddress() {
        return streetAddress;
    }

    public void setStreetAddress(String streetAddress) {
        this.streetAddress = streetAddress;
    }

    public int getPostalCode() {
        return postalCode;
    }

    public void setPostalCode(int postalCode) {
        this.postalCode = postalCode;
    }

    public String getMasterEmail() {
        return masterEmail;
    }

    public void setMasterEmail(String masterEmail) {
        this.masterEmail = masterEmail;
    }

    public String getMasterPassword() {
        return masterPassword;
    }

    public void setMasterPassword(String masterPassword) {
        this.masterPassword = masterPassword;
    }

    public ArrayList<Profile> getProfiles() {
        return profiles;
    }

    public void setProfiles(ArrayList profiles) {
        this.profiles = profiles;
    }



    // Custom methods
    public void addProfile(Profile profile) {
        this.profiles.add(profile);
    }
}

File: Fam.ly - iOS Edition/Fam.ly - iOS Edition/Objects/Account/Account.swift
//
//  Account.swift
//  Fam.ly - iOS Edition
//
//  Created by Johnny Mack on 5/25/19.
//  Copyright © 2019 John Mack. All rights reserved.
//

import Foundation
import UIKit

public class Account: Codable {
    
    
    
    
    // Class properties
    var familyName: String
    var streetAddress: String
    var postalCode: Int
    var masterEmail: String
    var masterPassword: String
    var parents: [Parent]
    var children: [Child]
    
    
    
    // Constructor
    init(FamilyName familyName: String, StreetAddress streetAddress: String, PostalCode postalCode: Int, MasterEmail masterEmail: String,
         MasterPassword masterPassword: String) {
        self.familyName = familyName
        self.streetAddress = streetAddress
        self.postalCode = postalCode
        self.masterEmail = masterEmail
        self.masterPassword = masterPassword
        self.parents = [Parent]()
        self.children = [Child]()
    }
    
    
    
    // Getters / Setters
    func getFamilyName() -> String {
        return familyName
    }
    
    func setFamilyName(familyName: String) {
        self.familyName = familyName
    }
    
    func getStreetAddress() -> String {
        return streetAddress
    }
    
    func setStreetAddress(streetAddress: String) {
        self.streetAddress = streetAddress
    }
    
    func getPostalCode() -> Int {
        return postalCode
    }
    
    func setPostalCode(postalCode: Int) {
        self.postalCode = postalCode
    }
    
    func getMasterEmail() -> String {
        return masterEmail
    }
    
    func setMasterEmail(masterEmail: String) {
        self.masterEmail = masterEmail
    }
    
    func getMasterPassword() -> String {
        return masterPassword
    }
    
    func setMasterPassword(masterPassword: String) {
        self.masterPassword = masterPassword
    }
    
    func getParents() -> [Parent] {
        return parents
    }
    
    func setParents(parents: [Parent]) {
        self.parents = parents
    }
    
    func getChildren() -> [Child] {
        return children
    }
    
    func setChildren(children: [Child]) {
        self.children = children
    }
    
    
    
    // Cutom methods
    func addParent(parent: Parent) {
        self.parents.append(parent)
    }
    
    func addChild(child: Child) {
        self.children.append(child)
    }
}
